core/routemgmt/activateApi.js

```diff
@@ -0,0 +1,387 @@
+      body && console.log('addRouteToGateway: response body: '+JSON.stringify(body));
+
+      if (error) {
+        console.error('addRouteToGateway: Unable to configure the API Gateway: '+JSON.stringify(error))
+        reject('Unable to configure the API Gateway: '+JSON.stringify(error));
+      } else if (statusCode != 200) {
+        console.error('addRouteToGateway: Response code: '+statusCode)
+        reject('Unable to configure the API Gateway: Response failure code: '+statusCode);
+      } else if (!body) {
+        console.error('addRouteToGateway: Unable to configure the API Gateway: No response body')
+        reject('Unable to configure the API Gateway: No response received from the API Gateway');
+      } else {
+        resolve(body);
+      }
+    });
+  });
+}
+
+/**
+ * Update an existing document in database.
+ * @param cloudantDb  The DB object
+ * @param doc  The document to write.  It should have _id and _rev fields.
+ */
+function updateApiDocInDb(cloudantDb, doc) {
+  return new Promise( function(resolve, reject) {
+    cloudantDb.insert(doc, function(error, response) {
+      if (!error) {
+        console.log("success", response);
+        resolve(response);
+      } else {
+        console.log("error", JSON.stringify(error))
+        reject(error);
+      }
+    });
+  });
+}
+
+
+function makeGwApiDoc(api) {
+  var gwdoc = {};
+  gwdoc.basePath = api.apidoc.basePath;
+  gwdoc.name = api.apidoc.info.title;
+  gwdoc.resources = {};
+  if (api.gwApiGuid) {
+    gwdoc.id = api.gwApiGuid;
+  }
+  for (var path in api.apidoc.paths) {
+  console.log('Got dbapidoc path: ', JSON.stringify(path));
+    gwdoc.resources[path] = {};
+    var gwpathop = gwdoc.resources[path].operations = {};
+    for (var operation in api.apidoc.paths[path]) {
+      console.log('Got operation for path: ', operation);
+      console.log('Got operation backendMethod: ', api.apidoc.paths[path][operation]['x-ibm-op-ext'].backendMethod);
+      var gwop = gwpathop[operation] = {};
+      gwop.backendMethod = api.apidoc.paths[path][operation]['x-ibm-op-ext'].backendMethod;
+      gwop.backendUrl = api.apidoc.paths[path][operation]['x-ibm-op-ext'].backendUrl;
+      gwop.policies = api.apidoc.paths[path][operation]['x-ibm-op-ext'].policies;
+    }
+  }
+  return gwdoc;
+}
+
+function validateArgs(message) {
+  var tmpdoc;
+  if(!message) {
+    console.error('No message argument!');
+    return 'Internal error.  A message parameter was not supplied.';
+  }
+
+  if (!message.dbname) {
+    return 'dbname is required.';
+  }
+
+  if (!message.gwUrl) {
+    return 'gwUrl is required.';
+  }
+
+  if (!message.namespace) {
+    return 'namespace is required';
+  }
+
+  if (!message.basepath) {
+    return 'basepath is required';
+  }
+
+  return '';
+}
+
+function confidentialPrint(str) {
+    var printStr;
+    if (str) {
+        printStr = 'XXXXXXXXXX';
+    }
+    return printStr;
+}
```

Comment:
if the credential is not supplied (indicates a corrupt package), the log will show an empty string for the credential.  non-empty credentials are not logged.